### PR TITLE
API-5841 - pull examples from schema

### PR DESCRIPTION
**Pull request by alexandervalencia**
This PR supports [API-5841](https://vajira.max.gov/browse/API-5841).

This enables LOAST to correctly pull example/examples from content/schema fields when they exist.

### CODE REVIEW

File: src/utilities/example-group/example-group.factory.ts [hunk 1]
```diff
@@ -61,14 +49,82 @@ class ExampleGroupFactory {
...
+
+          return Boolean(parameter.content[key].example);
+        }
+
+        return false;
+      })
+      .reduce((examples, parameter) => {
+        return {
+          ...examples,
+          [parameter.name]: parameter.example,
```
Comment (RockslideJr) [inline src/utilities/example-group/example-group.factory.ts:85]: what about when the example comes from the schema or content? The filter will return true, but this won't pull out the actual value

File: src/utilities/example-group/example-group.factory.ts [hunk 2]
```diff
@@ -8,25 +13,8 @@ class ExampleGroupFactory {
...
-    const nonRequiredExamplesWithExampleField = parameters
-      .filter((parameter) => !parameter.required && parameter.example)
-      .reduce((examples, parameter) => {
-        return {
-          ...examples,
-          [parameter.name]: parameter.example,
-        };
-      }, {});
+    const requiredExamples = this.getRequiredExamples(parameters);
+    const nonRequiredExamples = this.getNonRequiredExamples(parameters);
```
Comment (RockslideJr) [inline src/utilities/example-group/example-group.factory.ts:17]: this naming seems confusing. Is there a way to call out that these are just values from the `example` field?

Comment (michaelkimball) [reply src/utilities/example-group/example-group.factory.ts:17]: From what I can tell, this is just a refactoring to put the logic into a function. It still pulls from both examples and example.

Comment (michaelkimball) [reply src/utilities/example-group/example-group.factory.ts:17]: nvmd

**Review by alexandervalencia — COMMENTED** [review]
@michaelkimball Looks good! Thanks for picking up the ticket and fixing it!

Unfortunately I can't approve but it has my recommendation for approval

**Review by ediiotero — APPROVED** [review]
LGTM 👍

**Pull request merged**